test/fixtures/requests/prepare-settings-no-signer-entries.json

```diff
@@ -0,0 +1,11 @@
+{
+  "signers": {
+    "threshold": 0,
+    "weights": [
+      {
+        "address": "r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59",
```

Comment:
Since we're deleting the list, why do we need to specify an address here at all?

Reply:
I was getting validation error when I left out the Weight and the address fields. As you suggested above, let me try making the `weights` field optional based on threshold value.

Reply:
I think you can make "weight" optional by removing it from the "required" array in `settings-plus-memos.json`: https://github.com/ripple/ripple-lib/blob/develop/src/common/schemas/objects/#L90

Reply:
@intelliot Thanks for your input. Instead of making the 'weight' field optional,
I made the 'weights' field optional(it has both weight and address) and updated the corresponding tests as well. 
I tested the following scenarios against the test net server:
1. When threshold: 0, and no weights are specified - The transaction is submitted successfully.
2. When threshold: 0 and optional 'weights' are passed - The transaction is submitted successfully.
3. When threshold > 0 and no 'weights' are passed - The transaction fails with a Rippled error.
4. When threshold > 0 and either weight or address is not specified - The transaction fails with 
a ValidationError.
Please let me know what you think about this approach.